### DISCUSSION TITLE
firefox: webrtc

### DISCUSSION THREAD

**Discussion by sdsddsd1** [kisslinux/repo#307] [Q&A]
Hello,
the current situation for firefox on wayland only is that webrtc has to be disabled. So no webcam or microfon through the browser.
I have seen that you have added the firefox-beta package with a webtrc patch.
Have you already tested the webcam functionality? I have build the package, but my webcam is still not recognized, though it should be working. I could successfully record through ffmpeg.
Greetings 👋

**Comment by dylanaraps** [comment]
I do not have a webcam so I cannot test it. I did test audio (and other functionality) however (which *seems* fine). This patch from upstream is *preliminary*, it's likely that there may be issues. 

Firefox ships a slightly older webrtc version (webrtc is an external component) so this may already be fixed in webrtc's upstream. Given the current webrtc situation (it's disabled), I applied this patch to hopefully improve things for users (with the best solution that exists *right now*). ie, it's better than nothing.

The relevant bugs to follow upstream are: 

- X11-less builds (includes webrtc): https://bugzilla.mozilla.org/show_bug.cgi?id=1661450
-  webrtc update: https://bugzilla.mozilla.org/show_bug.cgi?id=1654112

Do let me know if there are any other issues and I will do what I can.
